Modules/_asynciomodule.c

```diff
@@ -193,6 +193,49 @@ is_coroutine(PyObject *coro)
     return has_it;
 }
 
+/*[clinic input]
+_asyncio.isfuture
+
+    obj: object
+
+Return True if obj is a Future instance.
+
+This returns True when obj is a Future instance or is advertising
+itself as duck-type compatible by setting _asyncio_future_blocking.
+See comment in Future for more details.
+
+[clinic start generated code]*/
+
+static PyObject *
+_asyncio_isfuture_impl(PyObject *module, PyObject *obj)
+/*[clinic end generated code: output=7ceb4ecaca613a86 input=b417c209efff25ec]*/
+{
+    _Py_IDENTIFIER(__class__);
+    PyObject* class = _PyObject_GetAttrId(obj, &PyId___class__);
+    if (class == NULL) {
+      return NULL;
+    }
+    _Py_IDENTIFIER(_asyncio_future_blocking);
+    int class_has_attr = _PyObject_HasAttrId(
```

Comment:
Is it needed?

Reply:
__class__ may not have attr _asyncio_future_blocking
It's needed.